### ISSUE TITLE
CSystem extraction Issue

### ISSUE THREAD

**Issue by RedEye13** [arcusmaximus/VNTranslationTools#97]
Hello, while trying to decrypt the .a files of C System games I'm running into this issue 

![image](https://user-images.githubusercontent.com/30023402/209441703-a6285e09-96e0-400e-9096-8145cb8671e5.png)

What I tried so far is

1- I'm using the latest version of VNTools v0.0.25
2- used both CSystem Tools and GARbro to extract the files didn't affect the result
3- tried a second game that I want to work on same thing happened
4- according to GARbro both games are using Grand Guignol no Yoru Director's Cut for encryption (at least that's the only one I found to get the images to display correctly) several others would only work on some of the pictures 

Here's a link for the 2 games
https://vndb.org/v32745
https://vndb.org/v29958

Also I zipped the .a files for one of them hopefully it helps 
[input.zip](https://github.com/arcusmaximus/VNTranslationTools/files/10298790/input.zip)

**Comment by Cosetto** [comment]
Same issue with #95, it's not encrypted, the scripts are unicode string

**Comment by RedEye13** [comment]
Isn't this tool supposed to read Unicode strings? it always worked just fine with Kirikiri games that I tried in the past that had Unicode instead of Shift JS also should I close the issue if it's the same as #95?

**Comment by Cosetto** [comment]
Nope, it hasn't implied in Csystem script [link](https://github.com/arcusmaximus/VNTranslationTools/blob/main/VNTextPatch.Shared/Scripts/CSystemScript.cs)

**Comment by Cosetto** [comment]
Maybe this code can help to read and write unicode string
```
        public static string ReadWordLengthUnicodeString(this BinaryReader reader)
        {
            var length = reader.ReadInt16();
            var bytes = reader.ReadBytes(length);
            return Encoding.Unicode.GetString(bytes);
        }

        public static void WriteWordLengthUnicodeString(this BinaryWriter writer, string s)
        {
            var bytes = Encoding.Unicode.GetBytes(s);
            writer.Write(Convert.ToUInt16(bytes.Length));
            writer.Write(bytes);
        }
```

**Comment by onefive1156** [comment]
It seems to be encrypted
![image](https://user-images.githubusercontent.com/70904460/215391471-f14d9912-82ff-4ee8-b5bf-2d4ae31db129.png)
